test/json-test.js

```diff
@@ -7,18 +7,20 @@ var levelup = require('../lib/levelup.js')
 var leveldown = require('leveldown')
 var encDown = require('encoding-down')
 var async = require('async')
+var concat = require('concat-stream')
 var common = require('./common')
 var assert = require('referee').assert
 var refute = require('referee').refute
 var buster = require('bustermove')
 
-buster.testCase('JSON API', {
+buster.testCase('JSON encoding', {
   'setUp': function (done) {
     common.commonSetUp.call(this, function () {
       this.runTest = function (testData, assertType, done) {
         var location = common.nextLocation()
         this.cleanupDirs.push(location)
         levelup(encDown(leveldown(location), {
+          keyEncoding: 'json',
```

Comment:
Thanks to the added stream test, I found out that these tests were actually using serialized keys (`'[object Object]'`).

(this is not related to `deferred-leveldown`)